archivebox/core/mixins.py

```diff
@@ -10,10 +10,10 @@ def get_search_results(self, request, queryset, search_term: str):
 
         search_term = search_term.strip()
         if not search_term:
-            return qs, use_distinct
+            return qs.distinct(), use_distinct
         try:
             qsearch = query_search_index(search_term)
-            qs = qs | qsearch
+            qs = qs & qsearch
```

Comment:
```suggestion
            qs = qs | qsearch
```

reverting just this change for now so I can merge the `.distinct()` fix